tests/test_bcrypt.py

```diff
@@ -494,3 +498,41 @@ def test_2a_wraparound_bug():
         )
         == b"$2a$04$R1lJ2gkNaoPGdafE.H.16.1MKHPvmKwryeulRe225LKProWYwt9Oi"
     )
+
+
+@pytest.mark.thread_unsafe()
+def test_multithreading():
+    def get_id():
+        return uuid.uuid4().bytes
+
+    class User:
+        def __init__(self, id_, pw):
+            self.id_ = id_
+            self.salt = bcrypt.gensalt(4)
+            self.hash_ = bcrypt.hashpw(pw, self.salt)
+            self.key = bcrypt.kdf(pw, self.salt, 32, 50)
```

Comment:
I chose these parameters to make this run in a reasonable time. Still, it's a very slow test.